php/php-rr-sdk.md

```diff
@@ -0,0 +1,680 @@
+yield $this->sleep(1 * Time::DAY);
+```
+
+Can be combined with Promises:
+
+```php
+$a = Workflow::executeActivity(...);
+$t = Workflow::timer(1 * Time::MINUTE);
+
+yield Workflow::waitAny($a, $t);
+
+if ($t->isFired()) {
+    // ...
+}
+```
+
+> `$this->activitie` simply mocks activity object creation for simplicity of this proposal. 
+
+### Queries
+Workflow queries can be easily implemented using dedicated query method of workflow. The state of the workflow can be 
+extracted from the properties of the object.
+
+```php
+use Temporal\Workflow;
+
+class DemoWorkflow extends Workflow\Workflow
+{
+    private $step = 0;
+
+    public function queryStep(): int 
+    {
+        return $this->step;
+    }
+
+    public function run(string $input): string
+    {
+        yield $this->activities->doSomething($input);
+        $this->step++;
+
+        // ...
+    }
+}
+```
+
+> Query methods must be registered explicitly.
+
+An alternative approach can be based on registering the callback function during the workflow execution:
+
+```php
+Workflow::addQueryHandler('something', function() use (&$step) {
+    return $step;
+});
+```
+
+> Such an approach is more suitable for dynamic workflows. The principle of work is identical to pre-registered queries.
+
+#### Using Annotations
+For more advanced frameworks it should be possible to register query method using annotation or attribute (PHP8) as 
+in Java SDK.
+
+```php
+/**
+ * @Workflow\QueryMethod(name="something")
+ */
+public function querySomething(string $world): string 
+{
+    return 'hello';
+}
+```
+
+### Signals
+Unlike queries the signals are fist-class citizen of actual workflow code. Unlike activities the signal can be triggered
+at any moment, without prior expectation by workflow. 
+
+Signal method can be defined explicitly:
+
+```php
+class UploadWorkflow extends Workflow\Workflow
+{
+    /** @Temporal\SignalMethod(name="signal") */
+    public function handleSignal($input):void
+    {
+    
+    }
+
+    public function run(string $input): string
+    {
+        
+    }
+}
+```
+
+Or at runtime:
+
+```php
+public function run(string $input): string
+{
+    Workflow::addSignalHandler('name', function($input){
+    
+    });       
+
+    // ...
+}
+```
+
+> Need Temporal authors feedback.
+
+#### Handle Signals
+Incoming signals can be handled during any workflow `yield` call. Workflow should allow to explicitly wait for signal:
+
+```php
+public function run(string $input): string
+{
+    Workflow::addSignalHandler('name', function($input){
+        // do something
+    });       
+
+    yield new WaitSignal('name');
+}
+```
+
+### Logging
+The workflow logging does not differ from any other parts of the application, the only exception is the need to exclude
+logs triggered while workflow is replaying. We propose to use mocked PSR-3 logger for this purposes:
+
+```php
+$logger = new NonReplayingLogger(Workflow::ctx(), $parentLogger); // it is possible to omit ::ctx()
```

Comment:
I would omit the `ctx` here as the NonReplayingLogger constructor can access it directly.